src/che-openshift-authentication.ts

```diff
@@ -11,10 +11,26 @@
 import * as theia from '@theia/plugin';
 import * as che from '@eclipse-che/plugin';
 import { spawn } from 'child_process';
+import axios, { AxiosInstance } from 'axios';
+
+interface Attributes {
+    cluster: string
+}
+
+interface Data {
+    attributes: Attributes
+}
+
+interface OsUserResponse {
+    data: Data[]
+}
 
 export async function start(context: theia.PluginContext) {
     const machineToken = process.env['CHE_MACHINE_TOKEN'];
     const isMultiUser = !!(machineToken && machineToken.length > 0);
+    const axiosInstance: AxiosInstance = axios;
+    const cheApi = process.env['CHE_API'];
+    const isHostedChe = cheApi && cheApi.indexOf('https://che.openshift.io/api') !== -1;
```

Comment:
isn't it looking wired to hardcode an hosted platform there ?

Reply:
I mean, for example there is preview platform as well, etc, so maybe we need another way of saying it's an 'hosted che' ?

Reply:
Do you have any other ideas how to detect `Hosted Che`?